app/controllers/propertyrentals/income/IncomeSectionFinishedController.scala

```diff
@@ -0,0 +1,85 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.propertyrentals.income
+
+import controllers.actions._
+import forms.propertyrentals.income.IncomeSectionFinishedFormProvider
+import models.{JourneyContext, NormalMode}
+import navigation.Navigator
+import pages.propertyrentals.income.IncomeSectionFinishedPage
+import play.api.i18n.{I18nSupport, MessagesApi}
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import repositories.SessionRepository
+import service.JourneyAnswersService
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendBaseController
+import views.html.propertyrentals.income.IncomeSectionFinishedView
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class IncomeSectionFinishedController @Inject() (
+  override val messagesApi: MessagesApi,
+  sessionRepository: SessionRepository,
+  navigator: Navigator,
+  identify: IdentifierAction,
+  getData: DataRetrievalAction,
+  requireData: DataRequiredAction,
+  formProvider: IncomeSectionFinishedFormProvider,
+  val controllerComponents: MessagesControllerComponents,
+  view: IncomeSectionFinishedView,
+  journeyAnswersService: JourneyAnswersService
+)(implicit ec: ExecutionContext)
+    extends FrontendBaseController with I18nSupport {
+
+  val form = formProvider()
+
+  def onPageLoad(taxYear: Int): Action[AnyContent] = (identify andThen getData andThen requireData) {
+    implicit request =>
+      val preparedForm = request.userAnswers.get(IncomeSectionFinishedPage) match {
+        case None        => form
+        case Some(value) => form.fill(value)
+      }
+
+      Ok(view(preparedForm, taxYear))
+  }
+
+  def onSubmit(taxYear: Int): Action[AnyContent] = (identify andThen getData andThen requireData).async {
+    implicit request =>
+      form
+        .bindFromRequest()
+        .fold(
+          formWithErrors => Future.successful(BadRequest(view(formWithErrors, taxYear))),
+          value =>
+            for {
+              updatedAnswers <- Future.fromTry(request.userAnswers.set(IncomeSectionFinishedPage, value))
+              _              <- sessionRepository.set(updatedAnswers)
+              _ <- journeyAnswersService.setStatus(
+                     JourneyContext(
+                       taxYear = taxYear,
+                       mtditid = request.user.mtditid,
+                       nino = request.user.nino,
+                       journeyName = "rental-income"
+                     ),
+                     status = if (value) "completed" else "inProgress",
```

Comment:
Consider moving this text to a constants file to avoid duplication everywhere -  "completed" else "inProgress"

Reply:
Done!